File: sample/src/main/java/me/xiaopan/spear/sample/fragment/AppListFragment.java
package me.xiaopan.spear.sample.fragment;

import android.app.Activity;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.View;

import me.xiaopan.android.inject.InjectContentView;
import me.xiaopan.android.inject.InjectView;
import me.xiaopan.android.widget.PagerSlidingTabStrip;
import me.xiaopan.spear.sample.MyFragment;
import me.xiaopan.spear.sample.R;
import me.xiaopan.spear.sample.adapter.FragmentAdapter;

/**
 * App列表页面，用来展示已安装APP和本地APK列表
 */
@InjectContentView(R.layout.fragment_app_list)
public class AppListFragment extends MyFragment {
    @InjectView(R.id.pager_appList_content) private ViewPager viewPager;
    private GetAppListTagStripListener getPagerSlidingTagStripListener;
    private FragmentAdapter fragmentAdapter;

    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);

        if(activity instanceof GetAppListTagStripListener){
            getPagerSlidingTagStripListener = (GetAppListTagStripListener) activity;
        }else{
            getPagerSlidingTagStripListener = null;
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        if(getPagerSlidingTagStripListener != null){
            getPagerSlidingTagStripListener = null;
        }
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if(fragmentAdapter == null){
            Fragment[] fragments = new Fragment[2];
            fragments[0] = new InstalledAppFragment();
            fragments[1] = new AppPackageFragment();
            fragmentAdapter = new FragmentAdapter(getChildFragmentManager(), fragments);
        }
        viewPager.setAdapter(fragmentAdapter);
        getPagerSlidingTagStripListener.onGetAppListTabStrip().setViewPager(viewPager);
    }

    public interface GetAppListTagStripListener{
        PagerSlidingTabStrip onGetAppListTabStrip();
    }
}

File: sample/src/main/java/me/panpf/sketch/sample/ui/AppListFragment.kt
package me.panpf.sketch.sample.ui

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View
import kotlinx.android.synthetic.main.fm_pager.*
import me.panpf.adapter.pager.FragmentArrayPagerAdapter
import me.panpf.pagerid.PagerIndicator
import me.panpf.sketch.sample.R
import me.panpf.sketch.sample.base.BaseFragment
import me.panpf.sketch.sample.base.BindContentView

/**
 * App列表页面，用来展示已安装APP和本地APK列表
 */
@BindContentView(R.layout.fm_pager)
class AppListFragment : BaseFragment() {

    private val fragmentAdapter: FragmentArrayPagerAdapter by lazy {
        FragmentArrayPagerAdapter(childFragmentManager, arrayOf<Fragment>(InstalledAppFragment(), AppPackageListFragment()))
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        pagerFm_pager.adapter = fragmentAdapter
        (parentFragment as GetPagerIndicatorCallback).onGetPagerIndicator().setViewPager(pagerFm_pager)
    }

    interface GetPagerIndicatorCallback {
        fun onGetPagerIndicator(): PagerIndicator
    }
}
